lib/util/git/repo-store.ts

```diff
@@ -0,0 +1,250 @@
+      proc.stdin.write(input);
+      proc.stdin.end();
+    }
+
+    proc.stdout.on('data', (data) => {
+      stdout.push(data.toString());
+    });
+
+    proc.stderr.on('data', (data) => {
+      stderr.push(data.toString());
+    });
+
+    proc.on('close', (code) => {
+      resolve({
+        code,
+        stdout: stdout.join(''),
+        stderr: stderr.join(''),
+      });
+    });
+
+    proc.on('error', reject);
+  });
+}
+
+async function createBlob(content: string): Promise<LongCommitSha | null> {
+  const { code, stdout, stderr } = await execGit(
+    ['hash-object', '-w', '--stdin'],
+    content,
+  );
+
+  if (code !== 0) {
+    logger.debug(`git hash-object failed: ${stderr}`);
+    return null;
+  }
+
+  return stdout.trim() as LongCommitSha;
+}
+
+interface TreeEntry {
+  sha: string;
+  path: string;
+}
+
+async function createTree(entries: TreeEntry[]): Promise<LongCommitSha | null> {
+  const treeEntries = entries
+    .map(({ sha, path }) => `100644 blob ${sha}\t${path}`)
+    .join('\n');
+
+  const { code, stdout, stderr } = await execGit(['mktree'], treeEntries);
+
+  if (code !== 0) {
+    logger.debug(`git mktree failed: ${stderr}`);
+    return null;
+  }
+
+  return stdout.trim() as LongCommitSha;
+}
+
+interface CreateCommitOptions {
+  tree: LongCommitSha;
+  message: string;
+  parents?: LongCommitSha[];
+}
+
+async function createCommit({
+  tree,
+  message,
+  parents = [],
+}: CreateCommitOptions): Promise<LongCommitSha | null> {
+  const args = ['commit-tree', tree];
+
+  for (const parent of parents) {
+    args.push('-p', parent);
+  }
+
+  const { code, stdout, stderr } = await execGit(args, message);
+
+  if (code !== 0) {
+    logger.debug(`git commit-tree failed: ${stderr}`);
+    return null;
+  }
+
+  return stdout.trim() as LongCommitSha;
+}
+
+async function createRef(ref: string, sha: LongCommitSha): Promise<boolean> {
+  const { code, stderr } = await execGit(['update-ref', ref, sha]);
+
+  if (code !== 0) {
+    logger.debug(`git update-ref failed: ${stderr}`);
+    return false;
+  }
+
+  return true;
+}
+
+export async function deleteRef(ref: string): Promise<boolean> {
+  const { code, stderr } = await execGit(['update-ref', '-d', ref]);
+
+  if (code !== 0) {
+    logger.debug(`git update-ref failed: ${stderr}`);
+    return false;
+  }
+  return true;
+}
+
+export async function set(
+  key: string,
+  value: string,
+  blobName = 'data.json',
+): Promise<void> {
+  await syncGit();
+
+  const blobSha = await createBlob(value);
+  if (!blobSha) {
+    return;
+  }
+  logger.debug(`Git blob created: ${blobSha}`);
+
+  const treeSha = await createTree([{ sha: blobSha, path: blobName }]);
+  if (!treeSha) {
+    return;
+  }
+  logger.debug(`Git tree created: ${treeSha}`);
+
+  const commitSha = await createCommit({ tree: treeSha, message: key });
+  if (!commitSha) {
+    return;
+  }
+  logger.debug(`Git commit created: ${commitSha}`);
+
+  const refName = `refs/renovate-data/${key}`;
```

Comment:
Do you think we should separate `refs/renovate` from `refs/renovate-data`? e.g. not `refs/renovate/data/*`?

Reply:
Then we need to change the branches we do for platform-based commits

Reply:
Otherwise I guarantee we'll get a weird bug some day

Reply:
Yes, let's do that then. Have everything under `refs/renovate/<something>/*`

Reply:
BTW I thought you used `refs/renovate/branches/*` already

Reply:
Yes, but maybe let's have them under `refs/renovate/tmp/*`, to be clear they're subject to cleanup